packages/atomic/src/utils/accessibility-utils.ts

```diff
@@ -49,12 +51,79 @@ export class AriaLiveRegionController implements ReactiveController {
   }
 }
 
-//TODO: Reimplement to fit Lit
-export class FocusTargetController {
-  public setTarget(_el: HTMLElement) {}
-  public focusAfterSearch() {}
-  public focusOnNextTarget() {}
-  public async focus() {}
+export class FocusTargetController implements ReactiveController {
+  private bindings: AnyBindings;
+  private lastSearchId?: string;
+  private element?: HTMLElement;
+  private onFocusCallback?: Function;
+  private doFocusAfterSearch = false;
+  private doFocusOnNextTarget = false;
+
+  constructor(
+    private host: ReactiveControllerHost,
+    bindings: AnyBindings
+  ) {
+    this.host = host;
+    this.bindings = bindings;
+    this.host.addController(this);
+  }
+
+  public setTarget(el?: HTMLElement) {
+    if (!el) {
+      return;
+    }
+    this.element = el;
+    if (this.doFocusOnNextTarget) {
+      this.doFocusOnNextTarget = false;
+      this.focus();
+    }
+  }
+
+  public async focus() {
+    await defer();
+    this.element?.focus();
+    this.onFocusCallback?.();
+  }
+
+  public focusAfterSearch() {
+    this.lastSearchId = this.bindings.store.getUniqueIDFromEngine(
+      this.bindings.engine
+    );
+    this.doFocusAfterSearch = true;
+    return new Promise((resolve) => (this.onFocusCallback = resolve));
+  }
+
+  public focusOnNextTarget() {
+    this.doFocusOnNextTarget = true;
+    return new Promise((resolve) => (this.onFocusCallback = resolve));
+  }
+
+  public disableForCurrentSearch() {
+    if (
+      this.bindings.store.getUniqueIDFromEngine(this.bindings.engine) !==
+      this.lastSearchId
+    ) {
+      this.doFocusAfterSearch = false;
+    }
+  }
+
+  hostUpdated() {
+    if (
+      this.doFocusAfterSearch &&
+      this.bindings.store.getUniqueIDFromEngine(this.bindings.engine) !==
+        this.lastSearchId
+    ) {
+      this.doFocusAfterSearch = false;
+      if (this.element) {
+        const el = this.element;
+        // The focus seems to be flaky without deferring, especially on iOS.
```

Comment:
What happens without it?

Reply:
This is an old comment that I just moved over; I haven't personally experienced the flakiness

Reply:
I'm also adding a comment to indicate that this should be investigated following the Lit migration